### PR TITLE
[Select] Fix form bug

### PR DESCRIPTION
**Pull request by huntabyte**
Since we weren't specifying `type="button"` for the select trigger, it was catching the click event of the form, causing all sorts of weird issues.

Closes: #410

### CODE REVIEW

File: src/docs/previews/dialog/main/tailwind/index.svelte [hunk 1]
```diff
@@ -18,80 +19,4 @@
...
-				class="absolute right-4 top-4 inline-flex h-6 w-6 appearance-none
-                items-center justify-center rounded-full p-1 text-magnum-800
-                hover:bg-magnum-100 focus:shadow-magnum-400"
-			>
-				<X class="square-4" />
-			</button>
-		</div>
-	{/if}
-</div>
+<PortalNested />
```
Comment (TGlide) [inline src/docs/previews/dialog/main/tailwind/index.svelte:22]: Oops?

File: src/lib/builders/combobox/create.ts [hunk 2]
```diff
@@ -608,7 +608,14 @@ export function createCombobox<ItemValue>(props?: CreateComboboxProps<ItemValue>
 	/** ------------------- */
 
 	onMount(() => {
-		open.set(withDefaults.defaultOpen);
+		const $forceVisible = get(forceVisible);
```
Comment (TGlide) [inline src/lib/builders/combobox/create.ts:611]: Why is this in this PR?

File: src/lib/builders/select/create.ts [hunk 3]
```diff
@@ -149,12 +149,18 @@ export function createSelect<
 	const { typed, handleTypeaheadSearch } = createTypeaheadSearch();
 
 	onMount(() => {
+		const $forceVisible = get(forceVisible);
```
Comment (TGlide) [inline src/lib/builders/select/create.ts:152]: Same here

File: src/docs/previews/dialog/main/tailwind/index.svelte (hunk 1 above)
Comment (huntabyte) [reply src/docs/previews/dialog/main/tailwind/index.svelte]: Big oops

File: src/lib/builders/combobox/create.ts (hunk 2 above)
Comment (huntabyte) [reply src/lib/builders/combobox/create.ts]: Oops - moving too fast

File: src/lib/builders/select/create.ts (hunk 3 above)
Comment (huntabyte) [reply src/lib/builders/select/create.ts]: Oops - moving too fast

**Review by TGlide — APPROVED** [review]

**Pull request merged**